gdal_subsetter/transform.py

```diff
@@ -935,20 +916,19 @@ def subset2(self, tiffile, outputfile, bbox=None, band=None, shapefile=None):
         # RasterFormat = 'GTiff'
         tmpfile = f'{os.path.splitext(outputfile)[0]}-tmp.tif'
         if bbox or shapefile:
-            with OpenGDAL(tiffile) as ref_ds:
-                if bbox:
-                    shapefile_out = self.box2shapefile(tiffile, bbox)
-                    boxproj, _ = boxwrs84_boxproj(bbox, ref_ds)
-                else:
-                    shapefile_out = os.path.join(os.path.dirname(outputfile),
-                                                 'tmpshapefile')
-                    boxproj, proj, shapefile_out, geometryname = shapefile_boxproj(
-                        shapefile, ref_ds, shapefile_out
-                    )
+            if bbox:
+                shapefile_out = box_to_shapefile(tiffile, bbox)
+                boxproj, _ = boxwrs84_boxproj(bbox, tiffile)
+            else:
+                shapefile_out = os.path.join(os.path.dirname(outputfile),
+                                             'tmpshapefile')
+                boxproj = shapefile_boxproj(shapefile, tiffile,
+                                            shapefile_out)
```

Comment:
The additional outputs from this function were either not being used (`proj`, `geometryname`) or were already defined prior to the function call. So I simplified the output from `shapefile_boxproj` to just return what was needed.